src/Microsoft.AspNet.Server.Kestrel/Infrastructure/KestrelThread.cs

```diff
@@ -120,6 +126,18 @@ private void OnStopRude(object obj)
                 IntPtr.Zero);
         }
 
+        private void OnStopForceUnreference(object obj)
+        {
+            _engine.Libuv.walk(
+                _loop,
+                (ptr, arg) =>
+                {
+                    var handle = UvMemory.FromIntPtr<UvHandle>(ptr);
+                    _loop.Libuv.unref(handle);
```

Comment:
Not sure; though I found it was doing it when debugging. Best guess is its multi-dispose protection from by the SafeHandle, and they get added back somehow, but will only dispose once?